src/iexamine.cpp

```diff
@@ -288,12 +288,20 @@ bool iexamine::harvestable_now( const tripoint &examp )
 void iexamine::cvdmachine( Character &you, const tripoint & )
 {
     // Select an item to which it is possible to apply a diamond coating
+    std::array<std::reference_wrapper<const material_id>, 6> steels = {
+        material_steel,
+        material_lc_steel,
+        material_mc_steel,
+        material_hc_steel,
+        material_ch_steel,
+        material_qt_steel,
+    };
     item_location loc = g->inv_map_splice( []( const item & e ) {
```

Comment:
Need to capture `steels` in the lambda expression
```cpp
    item_location loc = g->inv_map_splice( [&steels]( const item & e ) {
```

Reply:
TY!

Reply:
It needs to be with an ampersand so it is captured by reference:

```cpp
[&steels]( const item & e ) { ... }
```

The meaning is different without the ampersand.
```cpp
[steels]( const item & e ) { ... }
```
It copies `steels` every time.